app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,126 @@
+<h1 class="text-center mb-2 text-xl font-extrabold text-gray-900  md:text-2xl lg:text-3xl">
+  <%= @recipe.name %>
+</h1>
+<div class="flex justify-center">
+<%= image_tag ("recipes.jpg"), :class => "w-6/12 border rounded-3xl"%>
+</div>
+<div class="flex flex-col gap-y-1">
+  <div class="flex justify-between">
+    <div class="text-xl font-semibold text-gray-900">
+      <span>Preparation Time: </span> <span class="italic"><%= @recipe.preparation_time %></span>
+    </div>
+    <%= form_for @recipe, method: :PATCH do |f| %>
+      <%= f.submit "Public", class: 'text-white bg-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm w-36 sm:w-auto px-3 py-2 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800' if @recipe.public %>
+      <%= f.submit "Private" , class: 'text-white bg-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm w-36 sm:w-auto px-3 py-2 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800' unless @recipe.public %>
+    <% end %>
+  </div>
+  <div class="text-lg font-semibold text-gray-900">
+    <span>Cooking Time: </span> <span class="italic"><%= @recipe.cooking_time %></span>
+  </div>
+  <div>
+    <span class="text-lg font-semibold text-gray-900">Steps :</span>
+    <span><%= @recipe.description %></span>
+  </div>
+  
+  <%# modal and its button %>
+  <% if can? :destroy, @recipe %>
+    <div class="flex justify-between my-2">
+      <!-- Modal toggle -->
+      <button data-modal-target="defaultModal" data-modal-toggle="defaultModal" class="block text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800" type="button">
+        Generate Shopping list
+      </button>
```

Comment:
- [ ] The generate shopping list button should open a modal to choose an inventory to create a shopping list with for that recipe as shown in the wireframe below. Currently, your button doesn't function at all, kindly fix that, please.
![Screenshot from 2023-02-17 10-40-43](https://user-images.githubusercontent.com/94551484/219609293-f53cd2b9-d853-4533-87e7-205cf2d2bf25.png)

- [ ] Also add a shopping list page to match the given wireframe below. This will display the shopping list generated with the recipe name, and inventory name, Shows the list of food that is missing by comparing the food in the recipe with the food in the inventory and Counts the total food items and total price of the missing food.
![image](https://user-images.githubusercontent.com/94551484/219609890-9e897367-e1d7-481c-8323-585b52996392.png)